lib/driver.js

```diff
@@ -182,15 +185,38 @@ class YouiEngineDriver extends BaseDriver {
     let iosArgs = {
       javascriptEnabled: true,
     };
-    let iosdriver = new IOSDriver(iosArgs);
+    let driverName = "IOSDriver";
+    if (opts.platformVersion) {
+      let majorVer = opts.platformVersion.toString().split(".")[0];
+      if (parseInt(majorVer, 10) >= 10) {
+        driverName = "XCUITestDriver";
+      }
+    }
+
+    let iosdriver;
+    if (driverName === "XCUITestDriver") {
+      logger.info("Selected driver: XCUITestDriver");
+      iosdriver = new XCUITestDriver(iosArgs);
+    } else {
+      logger.info("Selected driver: IOSDriver");
+      iosdriver = new IOSDriver(iosArgs);
+    }
```

Comment:
couldn't you move this up into the block above? Then you could drop the `driverName` variable (unless it's used elsewhere?) - move this if block into the block above, replacing the `driverName === "XCUITestDriver"` part with the version check you're doing above. The `let iosdriver;` line would have to be moved up too.